contracts/StablecoinConverter.sol

```diff
@@ -133,30 +133,21 @@ contract StablecoinConverter is EpochTokenLocker {
       * @param validUntils batchIds represnnting order's expiry
       * @param buyAmounts relative minimum amount of requested buy amounts
       * @param sellAmounts maximum amounts of sell token to be exchanged
-      * @return total number of `msg.senders` orders
+      * @return `orderIds` an array of indices in which `msg.sender`'s orders are included
       *
       * Emits an {OrderPlacement} event with all relevant order details.
       */
-    function placeValidFromOrder(
+    function placeValidFromOrders(
         uint16[] memory buyTokens,
         uint16[] memory sellTokens,
         uint32[] memory validFroms,
         uint32[] memory validUntils,
         uint128[] memory buyAmounts,
         uint128[] memory sellAmounts
-    ) public returns (uint256) {
-        for (uint i; i < buyTokens.length; i++) {
-            orders[msg.sender].push(Order({
-                buyToken: buyTokens[i],
-                sellToken: sellTokens[i],
-                validFrom: validFroms[i],
-                validUntil: validUntils[i],
-                priceNumerator: buyAmounts[i],
-                priceDenominator: sellAmounts[i],
-                usedAmount: 0
-            }));
-            emit OrderPlacement(
-                msg.sender,
+    ) public returns (uint256[] memory orderIds) {
+        orderIds = new uint256[](buyTokens.length);
```

Comment:
I think there should be the `requires` here. Ganache, for example reverts with `VM Exception while processing transaction: invalid opcode` if you do an out of bounds index. And there should probably be tests that check that nothing bad happens when the input is invalid (all the arrays aren't the same length).

Reply:
I think that order placement should be as cheap as possible gas wise.

In my point of view, since it throws anyways, there is not really a benefit with checking it, besides having a proper error message.

Even if it does not longer throw in the future version of solc and the values are just set ==0, it would break anything in the exchange, as people can anyways set order values to 0.

Reply:
I just checked on remix and the gas difference was very small, it added 104 gas in total. If you guys are OK with this, then so am I. However I do think that we should at least add a unit test that checks that the place order reverts when array lengths don't match so we can verify our assumptions as things change. I don't think that a test for each array OOB is needed though.

Reply:
Just checked the bytecode, and solidity generates array bounds checks for each array access and includes `INVALID` opcodes if the bounds checks fail. One thing worth mentioning is that `INVALID` actually consumes all the gas from the call (!) and does not refund.

Reply:
Okie fella. Not exactly sure what the consensus was here, but I have included a test that invalid opcode is rejected and left the contract without any require statements. I also don't think they are necessary, because they do not affect the contract's security.

Reply:
Well they behave differently. If the contract fails because of `INVALID` opcode, which happens when trying to access an array out of bounds, it will consume all the gas that was sent. Failing with a `REVERT`, if a `require` is included, refunds remaining gas.

Reply:
I don't have a strong opinion here, as we are talking just about a little bit of gas. But I would tend to not do any requires.

I also think that it is perfectly fine to punish a bad guy, who does an invalid request, a lot, instead of charging more from all the valid requests.

Reply:
I'm generally not a fan of having 6 arrays which all are implicitly expected to have the same length (and would prefer a single array where the orders are packed in bytes one after the other).

I do agree with Nick that making this requirement explicit using a require statement is probably worth the 100 gas (and not using all gas if the arrays don't add up is fair).

But not feeling strong enough about it to change it now.

Reply:
I would also prefer passing encoded bytes and will create a new issue for the backlog in order to do this if everyone is all good with that.